src/clarity-angular/emphasis/alert/alert.ts

```diff
@@ -10,20 +10,41 @@ import {
     Output
 } from "@angular/core";
 
+// providers
+import { AlertIconAndTypesService } from "./providers/icon-and-types-service";
+
 @Component({
     selector: "clr-alert",
+    providers: [ AlertIconAndTypesService ],
     templateUrl: "./alert.html"
 })
 export class Alert {
+    constructor(public iconService: AlertIconAndTypesService) {
+    }
+
     @Input("clrAlertSizeSmall") isSmall: boolean = false;
-    @Input("clrAlertType") type: string = "alert-info";
     @Input("clrAlertClosable") closable: boolean = true;
     @Input("clrAlertAppLevel") isAppLevel: boolean = false;
 
     @Input("clrAlertClosed") _closed: boolean = false;
     @Output("clrAlertClosedChange") _closedChanged: EventEmitter<boolean> = new EventEmitter<boolean>(false);
 
-    private alertTypes: string[] = [ "alert-info", "alert-warning", "alert-danger", "alert-success"];
+    @Input("clrAlertType")
+    set alertType(val: string) {
+        this.iconService.alertType = val;
+    }
+    get alertType(): string {
+        return this.iconService.alertType;
+    }
+
+    @Input("clrAlertIcon")
+    set alertIconShape(value: string) {
+        this.iconService.alertIconShape = value;
+    }
```

Comment:
This one I'm not going to change. We already have components where people have to use Clarity Icons. I will open a github issue to allow for full customization of putting whatever anyone wants in that container, but it _would_ involve a templateRef to stamp in the content ala Wizard page buttons or whatnot.

Right now, users can create a Clarity Icon with any html they want using `ClarityIcons.add()` and inject content that way.